### ISSUE TITLE
How to set the hyper parameters of  train_protonet.py file without pre-trained weight?

### ISSUE THREAD

**Issue by lijichang** [Sha-Lab/FEAT#12]
Thank you for your great work!
I tried to implement the prototypical network with Resnet12, BUT I was failed. After reading your work, I found your code implement it, so I tried to run your train_protonet.py in my server. However, it could not run it to about 76% (5-shot 5-way) presented in your paper without pre-trained  weight. Can you tell me how to set the the hyper-parameters of  train_protonet.py ?
Best Wishes.

**Comment by Han-Jia** [comment]
Hello, 

The pre-trained weights are important for training based on the ResNet backbone. It is hard to make ResNet work well without the pre-train stage.

Based on my experience, the ResNet backbone will over-fit on MiniImageNet. You can try to do data augmentation during meta-learning to get better results or increase the number of ways during meta training as in the ProtoNet paper.

**Comment by lijichang** [comment]
> Hello,
> 
> The pre-trained weights are important for training based on the ResNet backbone. It is hard to make ResNet work well without the pre-train stage.


> 
> Based on my experience, the ResNet backbone will over-fit on MiniImageNet. You can try to do data augmentation during meta-learning to get better results or increase the number of ways during meta training as in the ProtoNet paper.

**Comment by lijichang** [comment]
Thanks for your answer above！
Sorry, I closed the question carelessly.
I have tried to train the Resnet12 with Pre-trainted weight. However, its test result in the first epoch had  reached 77% but had not increased the performance in the later epochs. Could you tell me how to obtain the pretrained weight of resnet12 you given?

Thanks.

**Comment by Han-Jia** [comment]
Hello, 

For ProtoNet with the 1-Shot 5-Way task, please try lr = 0.001, T = 64; schedule= 10-0.2; you may get validation accuracy at about 0.6641 and test accuracy around 0.61.
